testing/test_junitxml.py

```diff
@@ -678,7 +681,41 @@ def test_pass():
             pass
     ''')
 
-    result = testdir.runpytest(f,
+    result = testdir.runpytest(
+        f,
         '--dist', 'each', '--tx', '2*popen',
         '--junitxml', testdir.tmpdir.join("test.xml"))
-    assert 'INTERNALERROR' not in str(result.stdout)
\ No newline at end of file
+    assert 'INTERNALERROR' not in result.stdout.str()
+
+
+def test_fancy_items_regression(testdir):
+    ' issue 1259'
+    testdir.makeconftest("""
+        import pytest
+        class FunItem(pytest.Item):
+            def runtest(self):
+                pass
+        class NoFunItem(pytest.Item):
+            def runtest(self):
+                pass
+
+        class FunCollector(pytest.File):
+            def collect(self):
+                return [
+                    FunItem('a', self),
+                    NoFunItem('a', self),
+                ]
+
+        def pytest_collect_file(path, parent):
+            return FunCollector(path, parent)
+    """)
+
+    testdir.makepyfile('''
+        def test_pass():
+            pass
+    ''')
+
+    result = testdir.runpytest(
+        '--junitxml', testdir.tmpdir.join("test.xml"))
+
+    assert 'INTERNALERROR' not in result.stdout.str()
```

Comment:
Could we test the actual report contents instead of only checking that there wasn't an internal error? In particular, I would like the test to specify how the duplicated node ids appear on the XML